lib/bundler/source/git.rb

```diff
@@ -304,9 +304,9 @@ def git_proxy
 
       def fetch
         git_proxy.checkout
-      rescue GitError
+      rescue GitError => e
         raise unless Bundler.feature_flag.allow_offline_install?
-        Bundler.ui.warn "Using cached git data because of network errors"
+        Bundler.ui.warn "Using cached git data because of network errors\n#{e}"
```

Comment:
Maybe add a colon after `errors` to suggest the exception error contains details.